### PR TITLE
estimator: allow export_outputs to be a Tensor

### PR DESCRIPTION
**Pull request by boeddeker**
When someone wants to use `tf.estimator.Estimator.export_savedmodel` to save the predictions of a model and `tf.contrib.predictor.from_saved_model` to load the prediction function, it is currently necessary to define boilerplate code. This PR reduces the amount of code.

Currently:
```python
...
ret['export_outputs'] = {
        'predictions': tf.estimator.export.PredictOutput(
            {'predictions': ret['predictions']}
        )
    }
tf.estimator.EstimatorSpec(**ret)
```
With this PR the following is enough:
```python
ret['export_outputs'] = {'predictions': ret['predictions']}
ret['export_outputs'] = ret['predictions']
```
The predictor repr for the first example is equal to the original code
```python
SavedModelPredictor with feed tensors {'x': <tf.Tensor 'Placeholder:0' shape=(1, 10) dtype=float32>} and fetch_tensors {'predictions': <tf.Tensor 'dense/Relu:0' shape=(1, 10) dtype=float32>}
```
and for the second the output name changes
```python
SavedModelPredictor with feed tensors {'x': <tf.Tensor 'Placeholder:0' shape=(1, 10) dtype=float32>} and fetch_tensors {'output': <tf.Tensor 'dense/Relu:0' shape=(1, 10) dtype=float32>}
```

When someone wants to verify the code, here a full example:
```python
from types import SimpleNamespace
import numpy as np
import tensorflow as tf
from tensorflow.contrib import predictor

def model_fn(features, labels, mode):
    assert labels is None, labels
    x = features['x']
    y = features['y']
    
    y_hat = tf.layers.Dense(10, tf.nn.relu)(x)
    loss = tf.losses.mean_squared_error(y, y_hat)
    train_op = tf.train.AdamOptimizer(learning_rate=0.001).minimize(loss)
    
    ret = dict(
        mode=mode,
        loss=loss,
        predictions=y_hat,
        train_op=train_op,
    )

    # Boilderplate:
    ret['export_outputs'] = {  # <------------------- old
        'predictions': tf.estimator.export.PredictOutput(
            {'predictions': ret['predictions']}
        )
    }
    ret['export_outputs'] = {'predictions': ret['predictions']}  # <------------------- new
    ret['export_outputs'] = ret['predictions']  # <------------------- new alternate
    
    return tf.estimator.EstimatorSpec(**ret)
        
estimator = tf.estimator.Estimator(
    model_fn=model_fn
)

def generator():
    x = np.random.randn(1, 10).astype(np.float32)
    y = np.random.randn(1, 10).astype(np.float32)
    yield {'x': x, 'y': y}

def input_fn():
    ds = tf.data.Dataset.from_generator(generator, {'x': tf.float32, 'y': tf.float32}, {'x': [1, 10], 'y': [1, 10]})
    it = ds.make_one_shot_iterator()
    element = it.get_next()
    return element

estimator.train(input_fn)

def serving_input_receiver_fn():
    x = tf.placeholder(tf.float32, [1, 10])
    y = tf.placeholder(tf.float32, [1, 10])
    
    ret = SimpleNamespace()
    ret.features = {'x': x, 'y': y}  # for graph def
    ret.receiver_tensors = {'x': x}  # for serving
    ret.receiver_tensors_alternatives = None
    return ret


model_path = estimator.export_savedmodel('tmp', serving_input_receiver_fn)

predict_fn = predictor.from_saved_model(model_path)
# predict_fn({'x': np.random.randn(1, 10)})
predict_fn
```

### CODE REVIEW

**Comment by drpngx** [comment]
@davidsoergel WDYT?

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: It has been 16 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: It has been 151 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: It has been 166 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: It has been 181 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: It has been 196 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 211 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 226 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 241 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 256 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 271 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 286 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 301 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 316 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 331 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @davidsoergel: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 346 days with no activity and the `awaiting review` label has been applied.

**Comment by ymodak** [comment]
@boeddeker Thanks for your contribution and apologies for the delay in reverting. Can you please sync with latest TensorFlow branch and test again? @martinwicke Can you please take a look? Thanks!

**Review by martinwicke — CHANGES_REQUESTED** [review]
This is a nice usability improvement. Can you add a unit test to make sure this works and keeps working? It should exercise all branches of the new code.

**Comment by boeddeker** [comment]
@ymodak and @martinwicke: I can not rebase to the current master, because the code is moved to `tensorflow_estimator`.
Furthermore, I can not run the tests in `tensorflow_estimator`, so it is difficult for me to write tests.
The reason for that may be that tensorflow 1.12 does not use `tensorflow_estimator` (I use pip to install tensorflow).

**Comment by martinwicke** [comment]
@case540 is tensorflow_estimator installed separately, or bundled using bazel? I thought it was the former, so installing tensorflow, you should have tensorflow_estimator.

Either way, it doesn't matter much, since you created this patch as well, without the tensorflow source being installed via pip -- you probably cloned the tensorflow repo, and copied your changes into the cloned tree, right?

Simply do the same thing with tensorflow_estimator. 

As to tests: most tests are runnable by just running the script. For final verification, we can trigger the tests on the PR you submit. While it is ideal if you can test in the same environment (using bazel and all), this isn't necessary.

**Comment by case540** [comment]
TensorFlow is installed separately. You may have to "pip install tensorflow_estimator", or "pip install tf-estimator-nightly" in order to run some TensorFlow tests (some contrib/ tests still depend on estimator).

Just cloning TensorFlow git repo and building with Bazel will not get you the tensorflow_estimator code.

Sorry for slow repsonse

**Comment by rthadur** [comment]
Closing this PR as the code is being moved to Estimator repo.
